agent/acs/handler/acs_handler.go

```diff
@@ -327,6 +273,122 @@ func startACSSession(ctx context.Context, client wsclient.ClientServer, timer tt
 	}
 }
 
+// handleACSError handles an error from ACS. It returns true if the session can reconnect
+// to ACS. Else, it returns false. It also applies the appropriate backoff policy for
+// errors which can result in the handler reconnecting to ACS. For an "InactiveInstance"
+// error, it emits an event to the deregister instance event stream.
+func (s *Session) handleACSError(acsError error, connectToACS chan<- struct{}) bool {
```

Comment:
right now its being used in tests. the idea was to do different things based on the return value, but the function itself is handling that right now.